### PR TITLE
Improve Garmin sync

### PR DESCRIPTION
**Pull request by aleung**
Major changes:
- Use asyncio, remove multithread, limit concurrent downloads to avoid exhaust connection pool or hit server limitation
- Add loading configuration from file (config file is optional, command line parameters still take precedence)
- Remove unused code
- Format python code with ~~yapf~~ black

### CODE REVIEW

**Comment by yihong0618** [comment]
@aleung I will take a look tonight.
But for the last part, I choose black could you tell me why use tapfer?

**Comment by aleung** [comment]
I didn't know you already use black. Let me reformat it with black and push a new commit.

**Comment by yihong0618** [comment]
@aleung 
Thank you very much for your suggestion and pr
Hi~ Just add you as Collaborator.

Can we use `httpx` instead of `aiohttp` in your new garmin_sync
Because I use `httpx` in upstream repo  [Runtastic](https://github.com/yihong0618/Runtastic) 
And in the running_page for runtastic_sync the code is not so simple, My plan is to copy the code to the runtastic_sync.py,
In that I also use async too. 
Any suggestion?

File: scripts/garmin_sync.py [hunk 1]
```diff
@@ -341,64 +200,75 @@ def __init__(self, status):
...
+
+    return await asyncio.gather(*(sem_task(task) for task in tasks))
+
+
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("email", help="email of garmin")
-    parser.add_argument("password", help="password of garmin")
+    parser.add_argument("email", nargs="?", help="email of garmin")
+    parser.add_argument("password", nargs="?", help="password of garmin")
```
Comment (yihong0618) [inline scripts/garmin_sync.py:238]: why nargs here?

File: scripts/garmin_sync.py [hunk 2]
```diff
@@ -341,64 +200,75 @@ def __init__(self, status):
...
+        traceback.print_exc()
         pass
 
 
+async def get_activity_id_list(client, start=0):
+    activities = await client.get_activities(start, 100)
+    if len(activities) > 0:
+        ids = list(map(lambda a: a.get("activityId"), activities))
+        logger.debug(f"Activity IDs: {ids}")
+        return ids + await get_activity_id_list(client, start + 100)
```
Comment (yihong0618) [inline scripts/garmin_sync.py:220]: cool, I use it in [runtastic](https://github.com/yihong0618/Runtastic/blob/master/runtastic.py#L259) too~

**Comment by aleung** [comment]
I don't use Python too much and never used aiohttp nor httpx before. I just googled and picked one to solve the too many concurrent download issue.

Have switched to httpx. Their API are similar so just little code change.

File: scripts/garmin_sync.py (hunk 1 above)
Comment (aleung) [reply scripts/garmin_sync.py:238]: I expect that command line arguments can be omitted when all needed parameters are available in config file. 

Do you agree with the use of configuration file? I know that people don't need it in GitHub action. I use this tool in non-public environment. If not using config file, I can write a shell script to populate my own settings.

**Comment by yihong0618** [comment]
@aleung merge first, I will test later, Thank you very much.

**Comment by yihong0618** [comment]
@aleung 
Works fine, thank you again, but I will do a little change for support Python3.6
because asyncio.run() is a new API for Python3.7

**Pull request merged**